options.go

```diff
@@ -263,16 +303,24 @@ func (o *Options) Validate() error {
 			}
 			keySet := oidc.NewRemoteKeySet(ctx, o.OIDCJwksURL)
 			o.oidcVerifier = oidc.NewVerifier(o.OIDCIssuerURL, keySet, &oidc.Config{
-				ClientID: o.ClientID,
+				ClientID:        o.ClientID,
+				SkipIssuerCheck: o.InsecureOIDCSkipIssuerVerification,
 			})
 		} else {
 			// Configure discoverable provider data.
 			provider, err := oidc.NewProvider(ctx, o.OIDCIssuerURL)
 			if err != nil {
+				// go-oidc doesn't let us pass bypass the issuer check this in the oidc.NewProvider call,
+				// so we'll catch this case and just warn that you have to manually provide it instead of
+				// rely on discovery.
+				if o.InsecureOIDCSkipIssuerVerification && strings.HasPrefix(err.Error(), "oidc: issuer did not match the issuer returned by provider") {
+					logger.Printf("Your have specified InsecureOIDCSkipIssuerVerification, but OIDC discovery is not currently compatible with this setting. Please also use SkipOIDCDiscovery and set the requisite URLs manually to use this option.")
+				}
```

Comment:
Does this not suggest that the code above in the `if o.InsecureOIDCSkipIssuerVerification && !o.SkipOIDCDiscovery` block is redundant? 

I think our best option may just be to document that people should manually specify the options rather than using discovery when the issuer does not match, WDYT?

Reply:
I thought about that, but it seemed needlessly prescriptive. Discovery isn't hard to implement in this case.

In fact, I'd actually vote for just doing the discovery here 100% of the time. Because:
1. There are endpoints that go-oidc doesn't currently grab, such as the signout URL (in fact, it hides or ignores an annoying amount of information that discovery provides).
2. In some specific cases (again, annoyingly, AzureAD) providers recommend non-standard methods of discovery, such as by appending query strings like "?appId=12312312312312", which we can't do inside oidc-go.

I didn't want to fully replace the existing discovery with this PR though, so I left this as a compromise.

Reply:
I just saw that the discovery you implemented sets skip to true, so theoretically we should never get here right?

Reply:
Yeah, you're correct. I did this first and forgot to remove it after adding the discovery above. Will remove.